src/wui/load_or_save_game.cc

```diff
@@ -129,6 +131,21 @@ bool LoadOrSaveGame::compare_map_name(uint32_t rowa, uint32_t rowb) const {
 	return get_savegame(rowa).compare_map_name(get_savegame(rowb));
 }
 
+bool LoadOrSaveGame::check_replay_compatibility(const SavegameData& sd) {
+	if (filetype_ != FileType::kReplay || sd.is_directory() || !sd.errormessage.empty() ||
+	    (SDL_GetModState() & KMOD_CTRL) != 0 || sd.version == build_id()) {
+		return true;
+	}
```

Comment:
This looks wrong to me. Isn't it supposed to be

```suggestion
bool LoadOrSaveGame::check_replay_compatibility(const SavegameData& sd) {
	if (filetype_ != FileType::kReplay || sd.is_directory() || !sd.errormessage.empty()) {
		return false;
	}
	if ((SDL_GetModState() & KMOD_CTRL) != 0 || sd.version == build_id()) {
		return true;
	}
```

?

Reply:
It's correct, but I added a detailed comment to clarify what this function does

Reply:
OK, thanks for the explanation.